rllib/agents/dqn/simple_q.py

```diff
@@ -215,3 +250,73 @@ def execution_plan(workers, config, **kwargs):
         )
 
         return StandardMetricsReporting(train_op, workers, config)
+
+    @ExperimentalAPI
+    def training_iteration(self) -> ResultDict:
+        """Simple Q training iteration function.
+
+        This replicates Simple Q's execution_plan behaviour, e.g.:
+        - For every time we (1) sample (MultiAgentBatch) from workers...
+            - (2) Concatenate freshly collected samples.
+            - (3) Store new samples in replay buffer.
+            - (4) Sample training batch (MultiAgentBatch) from replay buffer.
+            - (5) Learn on training batch.
+            - (6) Update target network every target_network_update_freq steps.
+        - (7) Return all collected metrics for the iteration.
+
+        Returns:
+            The results dict from executing the training iteration.
+        """
+        batch_size = self.config["train_batch_size"]
+        local_worker = self.workers.local_worker()
+
+        sample_batches = []
+        train_results = {}
+
+        # (1) Sample (MultiAgentBatch) from workers
+        new_sample_batches = synchronous_parallel_sample(self.workers)
```

Comment:
One more question on this: Why do we need `sample_batches` and `new_sample_batches`? Wouldn't one suffice?

Reply:
It is nonsense. It is a "leftover" from your [other request](https://github.com/ray-project/ray/pull/22842#discussion_r830856256). Thanks!